### ISSUE TITLE
Parallelizer: Spurious loop carried depencence error in inlined tasks

### ISSUE THREAD

**Issue by mariodirenzo** [StanfordLegion/legion#488]
Compiling this code
```
import "regent"

fspace MeshFields {
   x : double -- Unknowns of the implicit solver
}

__demand(__inline)
task rhs( Mesh : region(ispace(int3d), MeshFields),
          elem : int3d(MeshFields, Mesh) )
where
   reads writes (Mesh.x)
do
   var f =  (@elem).x
   return f
end

__demand (__inline)
task foo( Mesh : region(ispace(int3d), MeshFields),
           elem : int3d(MeshFields, Mesh) )
where
   reads writes(Mesh.x)
do
   (@elem).x += 1
   var dx1 = rhs(Mesh, elem)
end

__demand(__cuda, __leaf, __parallel)
task AdvanceMesh( Mesh : region(ispace(int3d), MeshFields) )
where
   reads writes(Mesh.x)
do
   __demand(__openmp)
   for c in Mesh do
      for step = 0, 1000 do
         -- This does not compile
         foo(Mesh, c)
         -- This does
         --Mesh[c].x += 1
         --var dx1 = rhs(Mesh, c)
      end
   end
end
```
produces the following error
```
Example2.rg:13: OpenMP code generation failed: found a loop-carried dependence
   var f =  (@elem).x
```

If the content of the task foo is copied inside the main loop, the error disappears.

**Comment by magnatelee** [comment]
Should be fixed by 147fbea7. Could you pull and try again?

**Comment by mariodirenzo** [comment]
Fixed. Thanks
